analyzers/dataframe/CMakeLists.txt

```diff
@@ -17,11 +17,29 @@ include_directories(${EDM4HEP_INCLUDE_DIRS}
                     ${VDT_INCLUDE_DIR}
                     )
 
-message(STATUS "includes-------------------------- dataframe awkward: ${AWKWARD_INCLUDE}")
-message(STATUS "includes-------------------------- dataframe edm4hep: ${EDM4HEP_INCLUDE_DIRS}")
-message(STATUS "includes-------------------------- dataframe podio  : ${podio_INCLUDE_DIR}")
-message(STATUS "includes-------------------------- dataframe fastjet: ${FASTJET_INCLUDE_DIRS}")
-
+message(STATUS "includes-------------------------- dataframe awkward:     ${AWKWARD_INCLUDE}")
+message(STATUS "includes-------------------------- dataframe edm4hep:     ${EDM4HEP_INCLUDE_DIRS}")
+message(STATUS "includes-------------------------- dataframe podio  :     ${podio_INCLUDE_DIR}")
+message(STATUS "includes-------------------------- dataframe fastjet:     ${FASTJET_INCLUDE_DIRS}")
+
+set(EXTRA_INCLUDE_DIRS)
+set(EXTRA_LIBRARIES)
+
+if(${WITH_ONNX})
+  find_package(ONNXRuntime)
+  find_package(nlohmann_json QUIET)
+  if(ONNXRuntime_FOUND AND nlohmann_json_FOUND)
+    message(STATUS "includes-------------------------- dataframe onnxruntime: ${ONNXRUNTIME_INCLUDE_DIRS}")
+    message(STATUS "includes-------------------------- dataframe nlohmann_json")
+    list(APPEND EXTRA_INCLUDE_DIRS ${ONNXRUNTIME_INCLUDE_DIRS})
+    list(APPEND EXTRA_LIBRARIES ${ONNXRUNTIME_LIBRARIES} nlohmann_json::nlohmann_json)
+    include_directories(${ONNXRUNTIME_INCLUDE_DIRS})
+  else()
+    message(WARNING "ONNXRuntime and/or nlohmann's JSON libraries not found.")
+    set(WITH_ONNX OFF)
```

Comment:
If using an explicit option, we don't want to turn it OFF if onnx is not there, but fail immediately. Otherwise, you might up with an incorrect installation of FCCAnalyses, which is worse than a failed build.

However, it's possible to add an `AUTO` mode to the `WITH_ONNX` that has the behavior you implement here and is probably a good default. We did this for clang-format in podio: https://github.com/AIDASoft/podio/blob/1f127616c4d6633471f8319a414610a4999015cd/cmake/podioMacros.cmake#L108

Reply:
Perfectly fine with me! Implemented this tristate.